lib/src/delegates/asset_picker_builder_delegate.dart

```diff
@@ -1404,29 +1410,38 @@ class DefaultAssetPickerBuilderDelegate
     BuildContext context,
     int index,
     List<AssetEntity> currentAssets, {
-    Widget? specialItem,
+    List<SpecialItemModel> specialItemModels = const [],
   }) {
     final DefaultAssetPickerProvider p =
         context.read<DefaultAssetPickerProvider>();
     final int length = currentAssets.length;
     final PathWrapper<AssetPathEntity>? currentWrapper = p.currentPath;
     final AssetPathEntity? currentPathEntity = currentWrapper?.path;
 
-    if (specialItem != null) {
-      if ((index == 0 && specialItemPosition == SpecialItemPosition.prepend) ||
-          (index == length &&
-              specialItemPosition == SpecialItemPosition.append)) {
-        return specialItem;
+    final prepandSpecialItemModels = specialItemModels.where(
+      (model) => model.position == SpecialItemPosition.prepend,
+    );
+    final appendSpecialItemModels = specialItemModels.where(
+      (model) => model.position == SpecialItemPosition.append,
+    );
```

Comment:
You may want to iterate the models once and `add` items accordingly to gain a better performance.

```suggestion
    final prependItems = <SpecialItemModel>[];
    final appendItems = <SpecialItemModel>[];
    for (final model in specialItemModels) {
      switch (model.position) {
        case SpecialItemPosition.prepend:
          prependItems.add(model);
        case SpecialItemPosition.append:
          appendItems.add(model);
      }
    }
```